test/python/Utils_t/PortForward_t.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+"""
+Unittests for PortForward
+"""
+
+from __future__ import division, print_function
+
+import unittest
+
+from Utils.PortForward import portForward, PortForward
+
+
+class RequestHandler(object):
+    def __init__(self, config=None, logger=None):
+        super(RequestHandler, self).__init__()
+        if not config:
+            config = {}
+
+    @portForward(8443)
+    def request(self, url, params=None, headers=None, verb='GET',
+                verbose=0, ckey=None, cert=None, doseq=True,
+                encode=False, decode=False, cookie=None, uri=None):
+        return url
+
+
+class PortForwardTests(unittest.TestCase):
+    """
+    Unittest for PortForward decorator and class
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(PortForwardTests, self).__init__(*args, **kwargs)
+        self.urlTestList = ['https://cmsweb.cern.ch/couchdb/reqmgr_workload_cache/_design/ReqMgr/_view/bydate?descending=true&limit=1',
+                            'https://cmsweb.cern.ch/couchdb/reqmgr_workload_cache/_design/ReqMgr/_view/bystatusandtime?startkey=%5B%22announced%22%2C+0%5D&endkey=%5B%22announced%22%2C+1616016936%5D&descending=false&stale=update_after&include_docs=false',
+                            'https://cmsweb.cern.ch/reqmgr2/js/?f=utils.js&f=ajax_utils.js&f=md5.js&f=task_splitting.js',
```

Comment:
Todor, to make sure we are well covered, what happens if the input url already contains the port number?
I.e. how does the unit test deal with an url like: `https://cmsweb.cern.ch:8443/reqmgr2/data/info`?

Reply:
It is working perfectly fine. If the link has any kind of a port suffix, then during the parsing step the port is separated from the base url and is substituted correctly with the one provided from the decorator call. So in case we have a link like:
`https://cmsweb.cern.ch:8443/reqmgr2/data/info` it will result in the very same link but only of the cost of few CPU cycles - the cost for port swapping after the url parsing. But when considering to put an `if` check somewhere in the beginning so I can spare those CPU cycles it went worse because one cannot avoid the initial parsing and then all the benefit when it comes to few already mangled urls will cast a repetition of the same 'if` for any other `url` beeing decorated. So I opted for paying the cheapest price, even though at first glance it seemed the other way around. 

Should I include few of those in the unittest so that we have them covered?

Reply:
Yes, it makes sense. Yes, I'd say it's better to add it as well (no need to add in all 3 string "styles" though).

Reply:
Done